src/widgets/VaccineChart.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { View } from 'react-native';
+
+import {
+  VictoryLine,
+  VictoryScatter,
+  VictoryChart,
+  VictoryAxis,
+  VictoryTheme,
+} from 'victory-native';
+
+import Svg from 'react-native-svg';
+
+import { HazardPoint } from './HazardPoint';
+
+export class VaccineChart extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      width: null,
+      height: null,
+    };
+  }
+
+  // Victory Native sizes are set using absolute values. Parents dimensions are used to
+  // calculate relative values for width and height for each chart.
+  onLayout = event => {
+    this.setState({
+      width: event.nativeEvent.layout.width,
+      height: event.nativeEvent.layout.height,
+    });
+  };
+
+  renderPlot() {
+    const { minLine, maxLine, hazards, minTemp, maxTemp } = this.props;
+    const { width, height } = this.state;
+
+    if (!width || !height) return null;
+
+    if (minLine.length + maxLine.length === 0) return null;
+
+    const dataKeys = { x: 'date', y: 'temp' };
+
+    const lines = [...minLine, ...maxLine];
+
+    const minDomain =
+      lines.reduce((reducedTemp, { temp }) => Math.min(reducedTemp, temp), minTemp) - 1;
+    const maxDomain =
+      lines.reduce((reducedTemp, { temp }) => Math.max(reducedTemp, temp), maxTemp) + 1;
+
+    const minTempPlotLine =
+      // minTemp default prop is Infinity.
+      minTemp !== Infinity ? (
+        <VictoryLine
+          y={() => minTemp}
+          style={{
+            data: { strokeDasharray: minLineStyles.strokeDasharray, stroke: minLineStyles.stroke },
+          }}
+        />
+      ) : null;
+
+    const maxTempPlotLine =
+      // maxTemp default prop is -Infinity.
+      maxTemp !== -Infinity ? (
+        <VictoryLine
+          y={() => maxTemp}
+          style={{
+            data: { strokeDasharray: maxLineStyles.strokeDasharray, stroke: maxLineStyles.stroke },
+          }}
+        />
+      ) : null;
+
+    const minLinePlotLine =
+      minLine.length > 0 ? (
+        <VictoryLine
+          data={minLine}
+          interpolation={minLineStyles.interpolation}
+          style={{ data: { stroke: minLineStyles.stroke } }}
+          x={dataKeys.x}
+          y={dataKeys.y}
+        />
+      ) : null;
+
+    const minLinePlotScatter =
+      minLine.length > 0 ? (
+        <VictoryScatter
+          data={minLine}
+          style={{ data: { fill: minLineStyles.fill } }}
+          x={dataKeys.x}
```

Comment:
@wlthomson:
  `x={dataKeys.x} y={dataKeys.y}` = `{...dataKeys}` in this case

Reply:
Good idea, fixed by 7e1b7740.